usecases/vessel-detection/index.md

```diff
@@ -0,0 +1,211 @@
+
+
+```python
+spatial_extent  = {
+          "west": 12.194377989297493,
+          "east": 12.758093271633888,
+          "south": 44.24420099164355,
+          "north": 44.85455845353388
+        }
+
+temporal_extent = [
+          "2021-10-01",
+          "2021-10-09"
+        ]
+
+collection      = "SENTINEL1_GRD"
+bands           = ["VV","VH"]
+
+s1_datacube = conn.load_collection(
+    collection,
+    spatial_extent=spatial_extent,
+    bands=bands,
+    temporal_extent=temporal_extent
+)
+```
+
+Load the geometries as a vector cube.
+
+
+```python
+sea_mask = s1_datacube.process("load_vector_cube", {"URL": SEA_MASK})
+```
+
+Replace the values that lie outside of our polygon with NaN values.
+
+
+```python
+masked_data = s1_datacube.mask_polygon(sea_mask)
+```
+
+Apply a window to our datacube, and per pixel, update the pixel's value to be the mean of the window surrounding it.
+
+This would be the equivalent of the `data.rolling()` from the xarray example above.
+
+
+```python
+aggregated_window = masked_data.aggregate_spatial_window(size=[30, 30], reducer="mean")
+```
+
+Apply the multiplication of our given value to the result of the spatial window. We multiply to increase the contrast between sea and ship, which makes the detection more reliable.
+
+
+```python
+multiplied = aggregated_window.apply(lambda x: x.multiply(4) )
+```
+
+We then compare the pixel values from the previous node, against the values of the data we initially masked. We want to keep any values from the initially masked data, in order to generate a boolean mask, which should reflect our vessels.
+
+I could not find a nice way to express this in the client, without it re-including the first three nodes of the process graph. So I have applied it directly as a process here.
+
+
+```python
+lt_comparison = multiplied.process("apply", {
+                    "data": multiplied,
+                    "context": masked_data,
+                    "process": {
+                      "process_graph": {
+                        "lt1": {
+                          "process_id": "lt",
+                          "arguments": {
+                            "x": {
+                              "from_parameter": "x"
+                            },
+                            "y": {
+                              "from_parameter": "context"
+                            }
+                          },
+                          "result": True
+                        }
+                      }
+                    }
+                  }
+                )
+```
+
+Convert the resulting boolean datacube to a vector cube, and output the result in save result as a GeoJson. This will make the comparison of our results with the Maritime traffic data more straight forward.
+
+
+```python
+output_data = lt_comparison.process("raster_to_vector", {"data": lt_comparison})
+```
+
+
+```python
+vessel_detection = output_data.save_result(format="GeoJSON")
```

Comment:
the next 20 lines can be replaced with an 'execute_batch', would focus the usecase more on the interesting bits.

Reply:
I'd opt just for removing it from the docs, will leave it in the corresponding notebook as I don't need the results to be downloaded locally

Reply:
ok!